modules/ROOT/pages/installing-kubernetes.adoc

```diff
@@ -0,0 +1,264 @@
++
+[source,console]
+----
+secret "sgw-config" created
+----
+
+=== Deployment controller
+
+In this section you will deploy the Sync Gateway cluster with the configuration file that you created above.
+
+You will be using a https://kubernetes.io/docs/concepts/workloads/controllers/deployment/[Kubernetes deployment controller].
+A deployment controller allows you to define the number of Sync Gateway replicas and other parameters.
+
+. Create a new file called *sgw-deployment.yaml* with the following.
++
+[source,yaml]
+----
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: sync-gateway <1>
+spec:
+  replicas: 2 <2>
+  template:
+    metadata:
+      labels:
+        app: sync-gateway
+    spec:
+      containers:
+        - name: sync-gateway
+          image: couchbase/sync-gateway:2.1.2-enterprise <3>
+          args: ["/sync-gateway-config/sgw-config.json"] <4>
+          volumeMounts:
+            - name: sgw-config-volume
+              mountPath: /sync-gateway-config
+              readOnly: true
+          env:
+            - name: GOMAXPROCS
+              value: "2"
+          resources:
+            requests:
+              cpu: "100m"
+            limits:
+              cpu: "100m"
+      volumes:
+        - name: sgw-config-volume
+          secret:
+            secretName: sgw-config
+----
+<1> `metadata.name`: The name of the deployment is "sync-gateway".
+<1> `spec.replicas`: 2 Sync Gateway replicas that are deployed at most.
+* For import node deployment, this *must* be 1 because there can be at most 1 Sync Gateway replicator in a cluster.
+If high availability is important, you probably want at least 2 Sync Gateway import nodes.
+* For replicator node deployment, this *must* be 1 because there can be at most 1 Sync Gateway import node in a cluster.
+<1> `containers[].image`: Points to the docker image for Sync Gateway.
+<1> `containers[].args`: Points to the Sync Gateway configuration file named "sgw-config-working.json" which is mounted at the path specified via the `volumeMounts` config.
+<1> `volumeMounts`: Specifies where to mount the volume into the container.
+<1> `volumes`: Specifies what to mount.
+In our case, the "secret" with name "sgw-config" corresponding to the Sync Gateway configuration that was created in the previous step is mounted.
+Learn more about Kubernetes volumes https://kubernetes.io/docs/concepts/storage/volumes/[here].
+. Deploy the Sync Gateway cluster using the deployment controller file.
++
+[source,console]
+----
+kubectl create -f sgw-deployment.yaml
+----
+If successful, you will see the following.
++
+[source,console]
+----
+deployment.extensions "sync-gateway" created
+----
+. You can check the status of the deployment with the following command until all the pods corresponding to the Sync Gateway are in the "Ready" state and the status is "Running".
++
+[source,console]
+----
+kubectl get pods
+----
++
+If successful, you will see the following.
```

Comment:
Do we need to clarify 'success criteria' here? The user may not see the cb-example etc depending on if they've deployed their Couchbase Server in the same namespace.

Reply:
```suggestion
If successful, you will see a listing of the Sync Gateway pods that were deployed. In the sample output below, we have Couchbase Server and Sync Gateway pods on the same cluster. In a production deployment, you will have Couchbase Server deployed on separate cluster and hence, will not be seen in the output
```

Reply:
Would have to be the same Kubernetes cluster, would just be in a different namespace

Reply:
```suggestion
If successful, you will see a listing of the Sync Gateway pods that were deployed.
In the sample output below, we have Couchbase Server and Sync Gateway pods running on the same namespace.
In a production deployment, you may have Couchbase Server deployed on a separate namespace.
```